cmd/operator-sdk/build/cmd.go

```diff
@@ -59,8 +61,15 @@ For example:
 		"Extra image build arguments as one string such as \"--build-arg https_proxy=$https_proxy\"")
 	buildCmd.Flags().StringVar(&imageBuilder, "image-builder", "docker",
 		"Tool to build OCI images. One of: [docker, podman, buildah]")
+
+	// Deprecated
 	buildCmd.Flags().StringVar(&goBuildArgs, "go-build-args", "",
```

Comment:
Don't even add the flag if we're in the new project layout.
```suggestion
if !projutil.IsNewOperatorLayout() {
	buildCmd.Flags().StringVar(&goBuildArgs, "go-build-args", "", ...)
}
```

But don't deprecate it for the old project layouts. Once all the KB integration is in place, we'll deprecate the entirety of the old project layout.